src/YWidget.cc

```diff
@@ -472,6 +474,7 @@ YWidget::getProperty( const std::string & propertyName )
     if ( propertyName == YUIProperty_VWeight		) return YPropertyValue( weight( YD_VERT  ) );
     if ( propertyName == YUIProperty_HStretch		) return YPropertyValue( stretchable( YD_HORIZ ) );
     if ( propertyName == YUIProperty_VStretch		) return YPropertyValue( stretchable( YD_VERT  ) );
+    if ( propertyName == YUIProperty_ID	&& this->hasId() ) return YPropertyValue(this->id()->toString());
```

Comment:
can it happen that widget does not have ID? Other properties does not have this condition.

Reply:
Yes, this is possible. The other properties handled on this level are just simple types like string, bool, int. The widget ID is a pointer to an object which might be 0; a widget might or might not have an ID. This is why this needs to be checked. hasId() simply checks of this->priv->_id is 0.